tests/test_api.py

```diff
@@ -205,29 +208,40 @@ def test_el_id(self, request, abort):
                     self.UUT.routes['/x-nmos/query/' + v + '/<ips_type>/<el_id>/']['GET'][0](t, EL_ID)
                 abort.assert_called_once_with(404)
 
+    @mock.patch('nmosquery.common.routes.make_response')
     @mock.patch('nmosquery.common.routes.abort', side_effect=AbortException)
     @mock.patch('nmosquery.common.routes.request')
-    def test_subscriptions_post(self, request, abort):
+    def test_subscriptions_post(self, request, abort, make_response):
         request.method = "POST"
         data = { 'foo' : 'bar', 'baz' : [ 'boop', ] }
+        args = []
+        test_obj = {"foo": "bar", "id": "test"}
 
         for v in API_VERSIONS:
+            path = '/x-nmos/query/' + v + '/subscriptions'
+
             request.get_data = mock.MagicMock(return_value=json.dumps(data))
+            make_response.reset_mock()
+            make_response.return_value.headers = {}
             self.queries[v].post_ws_subscribers.reset_mock()
-            self.queries[v].post_ws_subscribers.return_value = (mock.sentinel.obj, True)
-            self.assert_route_returns_value(v, '/x-nmos/query/' + v + '/subscriptions', [], (201, mock.sentinel.obj), request, method="POST")
+            self.queries[v].post_ws_subscribers.return_value = (test_obj, True)
+            self.assert_route_exists(path, request.method)
+            self.UUT.routes[path][request.method][0](*args)
             self.queries[v].post_ws_subscribers.assert_called_once_with(data)
 
             request.get_data = mock.MagicMock(return_value=json.dumps(data))
+            make_response.reset_mock()
+            make_response.return_value.headers = {}
             self.queries[v].post_ws_subscribers.reset_mock()
-            self.queries[v].post_ws_subscribers.return_value = (mock.sentinel.obj, False)
-            self.assert_route_returns_value(v, '/x-nmos/query/' + v + '/subscriptions', [], (200, mock.sentinel.obj), request, method="POST")
+            self.queries[v].post_ws_subscribers.return_value = (test_obj, False)
+            self.assert_route_exists(path, request.method)
+            self.UUT.routes[path][request.method][0](*args)
             self.queries[v].post_ws_subscribers.assert_called_once_with(data)
 
             request.get_data = mock.MagicMock(return_value="{")
```

Comment:
Should this have a closing bracket?

Reply:
I don't think so, that's just setting up to ensure an exception is thrown when invalid data is used.